packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -193,6 +207,56 @@ export class EntryControlCCAPI extends CCAPI {
 			EntryControlCommand.ConfigurationGet,
 		);
 
+		const valueDB = this.endpoint.getNodeUnsafe()!.valueDB;
+
+		// Validate key cache size
+		const minKeyCacheSize = valueDB.getValue<number>(
+			getMinKeyCacheSizeStateValueID(this.endpoint.index),
+		);
+		if (minKeyCacheSize !== undefined && keyCacheSize < minKeyCacheSize) {
+			throw new ZWaveError(
+				`The "keyCacheSize" (${keyCacheSize}) must be >= than "minKeyCacheSize" (${minKeyCacheSize})!`,
+				ZWaveErrorCodes.Argument_Invalid,
+			);
+		}
+
+		const maxKeyCacheSize = valueDB.getValue<number>(
+			getMaxKeyCacheSizeStateValueID(this.endpoint.index),
+		);
+		if (maxKeyCacheSize !== undefined && keyCacheSize > maxKeyCacheSize) {
+			throw new ZWaveError(
+				`The "keyCacheSize" (${keyCacheSize}) must be <= than "maxKeyCacheSize" (${maxKeyCacheSize})!`,
+				ZWaveErrorCodes.Argument_Invalid,
+			);
+		}
+
+		// Validate key cache timeout
+		const minKeyCacheTimeout = valueDB.getValue<number>(
+			getMinKeyCacheTimeoutStateValueID(this.endpoint.index),
+		);
+		if (
+			minKeyCacheTimeout !== undefined &&
+			keyCacheTimeout < minKeyCacheTimeout
+		) {
+			throw new ZWaveError(
+				`The "keyCacheTimeout" (${keyCacheTimeout}) must be >= than "minKeyCacheTimeout" (${minKeyCacheTimeout})!`,
+				ZWaveErrorCodes.Argument_Invalid,
+			);
+		}
+
+		const maxKeyCacheTimeout = valueDB.getValue<number>(
+			getMaxKeyCacheTimeoutStateValueID(this.endpoint.index),
+		);
+		if (
+			maxKeyCacheTimeout !== undefined &&
+			keyCacheTimeout > maxKeyCacheTimeout
+		) {
+			throw new ZWaveError(
+				`The "keyCacheTimeout" (${keyCacheTimeout}) must be <= than "maxKeyCacheTimeout" (${maxKeyCacheTimeout})!`,
+				ZWaveErrorCodes.Argument_Invalid,
+			);
+		}
```

Comment:
I would only check if a value is passed that lies outside what the specs allow. If the user provides a value the device does not report as supported, that is on him. The UIs already give enough guidance when you limit min/max with metadata.
That check should be inside the CC implementation, see https://github.com/zwave-js/node-zwave-js/pull/1838/files/efaee1bc81b088fdb8fbd7791cf23a718a1de1f9#diff-5e7c890d091030ea844ddf9bc6acf697c2883b79e41c86e3861e177b5bd49729